src/awe/rmdlarchive.cpp

```diff
@@ -0,0 +1,131 @@
+/* OpenAWE - A reimplementation of Remedys Alan Wake Engine
+ *
+ * OpenAWE is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * OpenAWE is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * OpenAWE is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenAWE. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+
+#include <src/common/memreadstream.h>
+#include "src/common/readstream.h"
+#include "src/common/exception.h"
+
+#include "rmdlarchive.h"
+#include "resman.h"
+
+static const uint32_t RMDL_MAGIC_ID = MKTAG('R', 'M', 'D', 'L');
+
+namespace AWE {
+
+RMDLArchive::RMDLArchive(Common::ReadStream &bin) : _stream(bin) {
+	load(_stream);
+}
+
+RMDLArchive::RMDLArchive(const std::string &resource) :
+		_rmdl(ResMan.getResource(resource)),
+		_stream(_rmdl ? *_rmdl : throw Common::Exception("Failed to load resource {}",resource)) {
+	load(_stream);
+}
+
+size_t RMDLArchive::getNumResources() const {
+	return _fileEntries.size();
+}
+
+Common::ReadStream *RMDLArchive::getResource(const std::string &rid) const {
+	for (const auto &entry : _fileEntries) {
+		if (entry.name == rid) {
+			byte *data = new byte[entry.size];
+
+			_stream.seek(entry.offset);
+			_stream.read(data, entry.size);
+
+			return new Common::MemoryReadStream(data, entry.size);
+		}
+	}
+
+	return nullptr;
+}
+
+void RMDLArchive::load(Common::ReadStream &bin) {
+	bin.seek(0);
+	const auto magic = bin.readUint32BE();
+	if (magic != RMDL_MAGIC_ID)
```

Comment:
Just a small hint, we use a rather uncommon const syntax name, which is preceeded by a 'k' which would be in this case kRMDL, since all those four character codes have the variable name corresponding to the four character code.

Reply:
Thanks for the hint. I renamed the constant. You also might want to revise `src/graphics/images/dds.cpp` because that's where I took the inspiration for the MAGIC_ID name.